docs/user-guide/kubeadm-reference.md

```diff
@@ -0,0 +1,126 @@
+Here we specify all the flags that can be used to customise the Kubernetes
+installation.
+
+- `--api-advertise-addresses` (multiple values are allowed)
+- `--api-external-dns-names` (multiple values are allowed)
+
+By default, `kubeadm init` will automatically detect IP addresses and use
+these to generate certificates for the API server. This uses the IP address
+of the default network interface. If you would like to access the API server
+through a different IP address, or through a hostname, you can override these
+defaults with `--api-advertise-addresses` and `--api-external-dns-names`.
+Specifying `--api-advertise-addresses` will disable auto detection.
+
+- `--cloud-provider`
+
+Currently, `kubeadm init` does not provide autodetection of cloud provider.
+This means that load balancing and persistent volumes are not supported out
+of the box. You can specify a cloud provider using `--cloud-provider`.
+Valid values are the ones supported by `controller-manager`, namely `"aws"`,
+`"azure"`, `"cloudstack"`, `"gce"`, `"mesos"`, `"openstack"`, `"ovirt"`,
+`"rackspace"`, `"vsphere"`. In order to provide additional configuration for
+the cloud provider, you should create a `/etc/kubernetes/cloud-config.json`
+file manually, before running `kubeadm init`. `kubeadm` will then automatically
+pick those settings up and ensure other nodes are configured correctly.
+
+- `--external-etcd-cafile` etcd certificate authority file
+- `--external-etcd-endpoints` (multiple values are allowed)
+- `--external-etcd-certfile` etcd client certificate file
+- `--external-etcd-keyfile` etcd client key file
+
+By default, `kubeadm` will deploy a single node etcd cluster on the master
+to store Kubernetes state. This means that any failure on the master node
+will require you to rebuild your cluster from scratch. Currently `kubeadm init`
+does not support automatic deployment of a highly available etcd cluster.
+If you would like to use your own etcd cluster, you can override this
+behaviour with `--external-etcd-endpoints`. `kubeadm` supports etcd client
+authentication using the `--external-etcd-cafile`, `--external-etcd-certfile`
+and `--external-etcd-keyfile` flags.
+
+- `--pod-network-cidr`
+
+By default, `kubeadm init` does not set node CIDR's for pods and allows you to
+bring your own networking configuration through a CNI compatible network
+controller addon such as [Weave Net](https://github.com/weaveworks/weave-kube),
+[Calico](https://github.com/projectcalico/calico-containers/tree/master/docs/cni/kubernetes/manifests/kubeadm)
+or [Canal](https://github.com/tigera/canal/tree/master/k8s-install/kubeadm).
+If you are using a compatible cloud provider or flannel, you can specify a
+subnet to use for each pod on the cluster with the `--pod-network-cidr` flag.
+This should be a minimum of a /16 so that kubeadm is able to assign /24 subnets
+to each node in the cluster.
+
+- `--service-cidr` (default '10.12.0.0/12')
```

Comment:
If you change this, the options have to be reflected in `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` for the DNS